File: ProjectEuler/Python/001.py
##
# @file 001.py
# @brief Calculates the sum of multiples (3, 5) below 1000 
# @author Deeno Burgan
# @version 1
# @date 2016-06-21

# Sum multiples of aMultiplier, only summing multiples less than aLimit
def sumMultiples(aMultiplier, aLimit):
    lSum = 0

    if( (aLimit % aMultiplier) == 0 ):
        # If it evenly divides, the the range is simply [1, oneMultipleBeforeLimit)
        lRange = range(1, int(aLimit / aMultiplier))
    else:
        # However, if it doesn't, then the range is [1, oneMultipleBeforeLimitPlusOne) 
        lRange = range(1, int( (aLimit + aMultiplier) / aMultiplier ))
 
    for lNumber in lRange:    
        lSum += (lNumber * aMultiplier) 

    return lSum

if __name__ == "__main__":
     lSum =  sumMultiples(5, 1000) + sumMultiples(3, 1000) - sumMultiples(15, 1000)
     print("Sum of multiples of 3 and 5 below 1000: ", lSum)  

File: ProjectEuler/CPP/001.cpp
/**
* @file 001.cpp
* @brief Calculates sum of multiples (3, 5) below 1000
* @author Deeno Burgan
* @version 1
* @date 2016-06-26
*/

#include <iostream>

using namespace std;

unsigned long int sumMultiples(unsigned long int aMultiplier, unsigned long int aLimit)
{
	unsigned long int lSum = 0.0;
	unsigned long int lUpperLim = 0.0;

	if( (aLimit % aMultiplier) == 0 )
		lUpperLim = aLimit / aMultiplier;
	else
		lUpperLim = (aLimit + aMultiplier) / aMultiplier;

	for(int lNumber = 0; lNumber < lUpperLim; lNumber++)
		lSum += lNumber * aMultiplier;

	return lSum;
}

int main(int argc, char* argv[])
{
	unsigned long int lSum = sumMultiples(5, 1000) + sumMultiples(3, 1000) - sumMultiples(15, 1000);
	cout << "Sum of multiples of 3 and 5 below 1000: " << lSum << endl;

	return 0;
}
